vim-engine/src/main/kotlin/com/maddyhome/idea/vim/vimscript/model/variables/Variable.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2003-2024 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.vimscript.model.variables
+
+import com.maddyhome.idea.vim.api.ExecutionContext
+import com.maddyhome.idea.vim.api.VimEditor
+import com.maddyhome.idea.vim.vimscript.model.VimLContext
+import com.maddyhome.idea.vim.vimscript.model.datatypes.VimDataType
+
+/**
+ * Base class for Vim variables
+ */
+abstract class Variable {
```

Comment:
Let's change this to interface, we don't have any implementations

Reply:
Done!